h2/connection.py

```diff
@@ -894,27 +895,25 @@ def send_data(self, stream_id, data, end_stream=False, pad_length=None):
         self.config.logger.debug(
             "Send data on stream ID %d with len %d", stream_id, len(data)
         )
-        frame_size = len(data)
+        data_size = len(data)
+        padding_per_frame = 0
         if pad_length is not None:
             if not isinstance(pad_length, int):
                 raise TypeError("pad_length must be an int")
             if pad_length < 0 or pad_length > 255:
                 raise ValueError("pad_length must be within range: [0, 255]")
             # Account for padding bytes plus the 1-byte padding length field.
-            frame_size += pad_length + 1
+            padding_per_frame = pad_length + 1
+        num_frames = math.ceil(data_size / float(self.max_outbound_frame_size))
+        data_size += padding_per_frame * num_frames
```

Comment:
So this set of changes is unnecessary: we don't split data up into multiple frames. That's considered a user-level concern.